airbyte-webapp/src/components/connectorBuilder/StreamTestingPanel/SchemaDiffView.tsx

```diff
@@ -0,0 +1,168 @@
+  const { editorView } = useConnectorBuilderFormState();
+  const [field, , helpers] = useField(`streams[${testStreamIndex}].schema`);
+  const formattedSchema = useMemo(() => inferredSchema && formatJson(inferredSchema), [inferredSchema]);
+
+  const [schemaDiff, setSchemaDiff] = useState<Diff>(() =>
+    editorView === "ui" ? getDiff(field.value, inferredSchema) : { changes: [], lossyOverride: false }
+  );
+
+  useDebounce(
+    () => {
+      if (editorView === "ui") {
+        setSchemaDiff(getDiff(field.value, inferredSchema));
+      }
+    },
+    250,
+    [field.value, inferredSchema, editorView]
+  );
+
+  return (
+    <FlexContainer direction="column">
+      {editorView === "ui" && field.value && field.value !== formattedSchema && (
+        <InfoBox icon={faWarning} className={styles.infoBox}>
+          <FlexItem grow>
+            <FlexContainer direction="column">
+              <FormattedMessage id="connectorBuilder.differentSchemaDescription" />
+              <FlexContainer>
+                <FlexItem grow>
+                  <Button
+                    full
+                    variant="dark"
+                    disabled={field.value === formattedSchema}
+                    onClick={() => {
+                      helpers.setValue(formattedSchema);
+                    }}
+                  >
+                    <FormattedMessage
+                      id={
+                        schemaDiff.lossyOverride
+                          ? "connectorBuilder.overwriteSchemaButton"
+                          : "connectorBuilder.useSchemaButton"
+                      }
+                    />
+                  </Button>
+                </FlexItem>
+                {schemaDiff.mergedSchema && schemaDiff.lossyOverride && (
+                  <FlexItem grow>
+                    <Button
+                      full
+                      variant="dark"
+                      onClick={() => {
+                        helpers.setValue(schemaDiff.mergedSchema);
+                      }}
+                    >
+                      <FormattedMessage id="connectorBuilder.mergeSchemaButton" />
+                    </Button>
+                  </FlexItem>
+                )}
+              </FlexContainer>
+            </FlexContainer>
+          </FlexItem>
+        </InfoBox>
+      )}
+      {!field.value && (
+        <Button
+          full
+          variant="secondary"
+          onClick={() => {
+            helpers.setValue(formattedSchema);
+          }}
+        >
+          <FormattedMessage id="connectorBuilder.useSchemaButton" />
+        </Button>
+      )}
+      <FlexItem>
+        {editorView === "yaml" || !schemaDiff.changes.length ? (
+          <pre className={styles.diffLine}>
+            {formattedSchema
+              .split("\n")
+              .map((line) => ` ${line}`)
+              .join("\n")}
+          </pre>
+        ) : (
+          schemaDiff.changes.map((change, changeIndex) => (
+            <pre
```

Comment:
We discussed switching from `<pre>` to Monaco editor for the Records/Request/Response display, so that users are able to collapse and expand sections of the JSON.

It seems like it will be harder to switch to Monaco for this schema diff view, because we are manually styling the different lines of the diff. Does that sound right to you? If so, I think it is probably fine to keep `<pre>` for just this view, since collapsing parts of the diff would probably be confusing anyway

Reply:
Maybe there is a way to do the per-line styling with Monaco as well, but I would solve that as part of the switch to Monaco.